homeassistant/components/mqtt/client.py

```diff
@@ -593,7 +593,7 @@ def _async_on_socket_close(
     def _async_writer_callback(self, client: mqtt.Client) -> None:
         """Handle writing data to the socket."""
         if (status := client.loop_write()) != 0:
-            self._async_on_disconnect(status)
+            self._async_handle_callback_exception(status)
```

Comment:
IMO it is not needed to explicitly call the on_disconnect callback here. Also the status is different from the `reason_code` from the V2 callbacks.